weave/integrations/google_genai/imagen_utils.py

```diff
@@ -0,0 +1,94 @@
+from functools import wraps
+from typing import Any, Callable, Optional
+
+import weave
+from weave.integrations.google_genai.gemini_utils import (
+    google_genai_gemini_postprocess_inputs,
+)
+from weave.trace.autopatch import OpSettings
+from weave.trace.weave_client import Call
+
+
+def google_genai_gemini_postprocess_outputs(
+    outputs: dict[str, Any],
+) -> list[dict[str, Any]]:
+    from io import BytesIO
+
+    from PIL import Image
+
+    modified_outputs = []
+    if hasattr(outputs, "generated_images"):
+        for image_data in outputs.generated_images:
+            modified_outputs.append(
+                {
+                    "image": {
+                        "gcs_uri": image_data.image.gcs_uri,
+                        "image_bytes": image_data.image.image_bytes,
+                        "mime_type": image_data.image.mime_type,
+                        "image": Image.open(BytesIO(image_data.image.image_bytes)),
```

Comment:
You should wrap the image load in a try except.  If it fails, you can return a placeholder  and let the user know the image was corrupted